AutoT4/ProjectItemSettings.cs

```diff
@@ -67,9 +67,24 @@ protected T Get<T>(T defaultValue = default(T), [CallerMemberName] string name =
             if (0 != _storage.GetItemAttribute(_itemId, Scope(name), out serializedValue))
                 return defaultValue;
 
-            return !String.IsNullOrWhiteSpace(serializedValue)
-                       ? (T)Convert.ChangeType(serializedValue, typeof(T))
-                       : defaultValue;
+            if (string.IsNullOrWhiteSpace(serializedValue))
+                return defaultValue;
+
+            try
+            {
+                if (typeof(T).IsEnum)
+                {
+                    return (T)Enum.Parse(typeof(T), serializedValue);
+                }
+                return (T)Convert.ChangeType(serializedValue, typeof(T));
+            }
+            catch
+            {
+                if (coercionFunc != null)
```

Comment:
Wouldn't it be better to put this just after the `if` on line 75? What if the coercion function throws?